internal/manifests/config/build.go

```diff
@@ -3,24 +3,88 @@ package config
 import (
 	"bytes"
 	"embed"
+	"fmt"
 	"html/template"
 	"io"
-)
-
-const (
+	"strings"
 
-	// TempoConfigFileName is the name of the config file in the configmap.
-	TempoConfigFileName = "config.yaml"
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+	"github.com/os-observability/tempo-operator/internal/manifests/manifestutils"
 )
 
 var (
 	//go:embed tempo-config.yaml
 	tempoConfigYAMLTmplFile embed.FS
 	tempoConfigYAMLTmpl     = template.Must(template.ParseFS(tempoConfigYAMLTmplFile, "tempo-config.yaml"))
+
+	//go:embed tempo-overrides.yaml
+	tempoTenantsOverridesYAMLTmplFile embed.FS
+	tempoTenantsOverridesYAMLTmpl     = template.Must(template.ParseFS(tempoTenantsOverridesYAMLTmplFile, "tempo-overrides.yaml"))
 )
 
-// Build builds a tempo configuration.
-func buildConfiguration(opts options) ([]byte, error) {
+func s3FromParams(params Params) s3 {
+	s3Endpoint := params.S3.Endpoint
+	s3Insecure := !strings.HasPrefix(s3Endpoint, "https://")
+	if strings.HasPrefix(s3Endpoint, "http://") {
+		s3Endpoint = strings.TrimPrefix(s3Endpoint, "http://")
+	} else {
+		s3Endpoint = strings.TrimPrefix(s3Endpoint, "https://")
+	}
+	return s3{
+		Endpoint: s3Endpoint,
+		Bucket:   params.S3.Bucket,
+		Insecure: s3Insecure,
+	}
+}
```

Comment:
Can we add a unittest for that? iam still not sure if it makes sense. Right now we always trim `https://` if the prefix is not `http://` right? So `abc://` would lead to a `https://` trim. As far as i can see, trim has no effect if there is nothing to trim. That means we dont need the else here, right?

Reply:
I would prefer to do it late, is not part of this PR :(

Reply:
```
// TrimPrefix returns s without the provided leading prefix string.
// If s doesn't start with prefix, s is returned unchanged.
```
ok, would it be ok for you to remove the if?
I think it can be replaced by the following.
```golang
s3Endpoint = strings.TrimPrefix(s3Endpoint, "https://")
s3Endpoint = strings.TrimPrefix(s3Endpoint, "http://")
```